### PR TITLE
Addressed static code analysis issues

### PR DESCRIPTION
**Pull request by bhroam**
#### Describe Bug or Feature
I ran cppcheck and clang's static code analyzer on the scheduler and addressed the findings.
[report.txt](https://github.com/openpbs/openpbs/files/6666160/report.txt)

Here is cppcheck's report.  I'm not attaching clang's because it was filled with false positives.  I only addressed the real issues.  Some of cppcheck's issues were false positives, but not many.

#### Describe Your Change
Addressed issues reported by static code analyzers

A lot of minor fixes.  I moved a lot of variables closer to their use.  There were 3 actual bugs:
1) If we compiled with DEBUG mode, we wouldn't chdir() into our priv dir
2) There was a path through the STF code where if a job needed to shrink close to its min_walltime, it could cause a FPE.
3) for the daemon service user, we called setuid() and setgid() without checking their return value.  It turned out that setgid() was failing.  I believe this is because we set the uid first, and then we no longer have the privilege to set the gid.  I now set the gid first. 

#### Attach Test and Valgrind Logs/Output
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16,UBUNTU2004
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|7916|4177|2|5|0|0|4170|

I reran the failed/error tests and they passed.

### CODE REVIEW

File: src/scheduler/resource_resv.cpp [hunk 1]
```diff
@@ -305,7 +303,7 @@ free_resource_resv_array(resource_resv **resresv_arr)
 }
 
 /**
- * @brief
+ * @brie&
```
Comment (vstumpf) [inline src/scheduler/resource_resv.cpp:306]: Is this intended?

File: src/scheduler/resource_resv.cpp [hunk 2]
```diff
@@ -2258,6 +2196,15 @@ selspec::selspec(selspec& oldspec)
 	defs = oldspec.defs;
 }
 
+selspec& selspec::operator=(selspec& oldspec)
```
Comment (vstumpf) [inline src/scheduler/resource_resv.cpp:2199]: this should be a const ref

**Review by vstumpf — CHANGES_REQUESTED** [review]
Some comments

**Review by arungrover — APPROVED** [review]
LGTM!

**Review by vstumpf — APPROVED** [review]
LGTM

**Pull request merged**